### PR TITLE
Fixes subvol delete on a non-btrfs volume

### LINKED ISSUE #42271: nuke-graph-directory.sh error on btrfs subvolume delete
**Description**
nuke-graph-directory.sh errors out at L66 of:
https://github.com/moby/moby/blob/2acab355ec7347b905b3e42bb6639d689cbea28a/contrib/nuke-graph-directory.sh#L59-L70
On non-btrfs systems. As this is called by post rm hooks in apt packaging the package fails to uninstall.

Specifically on my system which is using zfs; btrfs tools are installed by default, thus:
```
if command -v btrfs > /dev/null 2>&1; then
```
works, and 
```
find "$dir" -type d -inum 256
```
returns a number of directories which are not btrfs volumes on my system. 

**Steps to reproduce the issue:**
On ubuntu 20.10
1. sudo apt-get install docker.io
2. sudo apt-get purge docker.io

**Describe the results you received:**
```
Purging configuration files for docker.io (19.03.13-0ubuntu3) ...

Nuking /var/lib/docker ...
  (if this is wrong, press Ctrl+C NOW!)

+ sleep 10
ok
+ btrfs subvolume delete /var/lib/docker/containers/242b750a0952a99008adc6cd37f749423eeacc6c92dce2e3a509350977a41a8c
ERROR: Not a Btrfs filesystem: Invalid argument
dpkg: error processing package docker.io (--purge):
 installed docker.io package post-removal script subprocess returned error exit status 1
Errors were encountered while processing:
 docker.io
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

**Describe the results you expected:**
Command succeeds.

### PR DESCRIPTION
**Pull request by brettmilford**
Inode numbers are guaranteed to be unique only within a filesystem.
As such there is an edge case where these predicates are true on a
non-btrfs filesystem.

Closes #42271

Signed-off-by: Brett Milford <brettmilford@gmail.com>

**- What I did**
Added an additional check for filesystem type.

**- How to verify it**
Tested this code block on btrfs and non-btrfs (zfs) filesystems.

**- Description for the changelog**
Fixed issue where `nuke-graph-directory.sh` executes btrfs subvolume delete on a non-btrfs file.

### CODE REVIEW

**Review by thaJeztah — COMMENTED** [review]
left a minor nit; but let me ask @tianon @AkihiroSuda to review

File: contrib/nuke-graph-directory.sh [hunk 1]
```diff
@@ -61,10 +61,12 @@ if command -v btrfs > /dev/null 2>&1; then
 	# Source: http://stackoverflow.com/a/32865333
 	for subvol in $(find "$dir" -type d -inum 256 | sort -r); do
 		if [ "$dir" != "$subvol" ]; then
-			(
-				set -x
-				btrfs subvolume delete "$subvol"
-			)
+			if [ "$(stat -f --format=%T $subvol)" == "btrfs" ]; then
```
[the remark this answers is not in the document]
Comment (tianon) [reply contrib/nuke-graph-directory.sh:64]: Full agreement from me :innocent:

I'd go one step further, too:

```suggestion
			if [ "$(stat -f --format=%T "$subvol")" = "btrfs" ]; then
```

Or even (to catch any possible failing exit code from `stat`):

```suggestion
			if subvolType="$(stat -f --format=%T "$subvol")" && [ "$subvolType" = "btrfs" ]; then
```

File: contrib/nuke-graph-directory.sh [hunk 2]
```diff
@@ -61,10 +61,12 @@ if command -v btrfs > /dev/null 2>&1; then
 	# Source: http://stackoverflow.com/a/32865333
 	for subvol in $(find "$dir" -type d -inum 256 | sort -r); do
 		if [ "$dir" != "$subvol" ]; then
```
Comment (tianon) [inline contrib/nuke-graph-directory.sh:63]: Oh, it's probably worth bringing that conditinoal all the way up to here, ala:

```suggestion
		if [ "$dir" != "$subvol" ] && subvolType="$(stat -f --format=%T "$subvol")" && [ "$subvolType" = "btrfs" ]; then
```

(So we only have a single `if`)

Comment (brettmilford) [reply contrib/nuke-graph-directory.sh:63]: that's, nifty; and looks good to me.

**Review by tianon — APPROVED** [review]
LGTM :sweat_smile:

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**